functional/FunctionalTest.java

```diff
@@ -2846,9 +2846,43 @@ public static void getBucketLifecycle() throws Exception {
     try {
       client.makeBucket(MakeBucketArgs.builder().bucket(bucketName).build());
       try {
-        client.getBucketLifecycle(GetBucketLifecycleArgs.builder().bucket(bucketName).build());
+        LifecycleConfiguration config =
+            client.getBucketLifecycle(GetBucketLifecycleArgs.builder().bucket(bucketName).build());
+        if (config != null) {
+          throw new Exception("config: expected: <null>, got: <non-null>");
+        }
         testSetBucketLifecycle(bucketName);
-        client.getBucketLifecycle(GetBucketLifecycleArgs.builder().bucket(bucketName).build());
+        config =
+            client.getBucketLifecycle(GetBucketLifecycleArgs.builder().bucket(bucketName).build());
+        if (config == null) {
+          throw new Exception("config: expected: <non-null>, got: <null>");
+        }
+        List<LifecycleRule> rules = config.rules();
+        if (config.rules().size() != 1) {
+          throw new Exception("config.rules().size(): expected: 1, got: " + config.rules().size());
+        }
+        LifecycleRule rule = rules.get(0);
+        if (rule.status() != Status.ENABLED) {
+          throw new Exception(
+              "rule.status(): expected: " + Status.ENABLED + ", got: " + rule.status());
+        }
+        if (rule.expiration() == null) {
+          throw new Exception("rule.expiration(): expected: <non-null>, got: <null>");
+        }
+        if (rule.expiration().days() != 365) {
+          throw new Exception(
+              "rule.expiration().days(): expected: 365, got: " + rule.expiration().days());
+        }
+        if (rule.filter() == null) {
+          throw new Exception("rule.filter(): expected: <non-null>, got: <null>");
+        }
+        if (!"logs/".equals(rule.filter().prefix())) {
+          throw new Exception(
+              "rule.filter().prefix(): expected: logs/, got: " + rule.filter().prefix());
+        }
```

Comment:
It might be better to introduce an `equals` method on the `LifecycleConfiguration` class and use that to confirm that the config that is received on getting the lifecycle is equal to the config that was passed when setting it.

Reply:
As we would need to make a next major release, we can add this in a separate PR after the release.